flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliClient.java

```diff
@@ -389,51 +395,58 @@ private void validate(Operation operation, ExecutionMode executionMode) {
     private void callOperation(Operation operation, ExecutionMode mode) {
         validate(operation, mode);
 
-        if (operation instanceof QuitOperation) {
-            // QUIT/EXIT
-            callQuit();
-        } else if (operation instanceof ClearOperation) {
-            // CLEAR
-            callClear();
-        } else if (operation instanceof HelpOperation) {
-            // HELP
-            callHelp();
-        } else if (operation instanceof SetOperation) {
-            // SET
-            callSet((SetOperation) operation);
-        } else if (operation instanceof ResetOperation) {
-            // RESET
-            callReset((ResetOperation) operation);
-        } else if (operation instanceof CatalogSinkModifyOperation) {
-            // INSERT INTO/OVERWRITE
-            callInsert((CatalogSinkModifyOperation) operation);
-        } else if (operation instanceof QueryOperation) {
-            // SELECT
-            callSelect((QueryOperation) operation);
-        } else if (operation instanceof ExplainOperation) {
-            // EXPLAIN
-            callExplain((ExplainOperation) operation);
-        } else if (operation instanceof BeginStatementSetOperation) {
-            // BEGIN STATEMENT SET
-            callBeginStatementSet();
-        } else if (operation instanceof EndStatementSetOperation) {
-            // END
-            callEndStatementSet();
-        } else if (operation instanceof AddJarOperation) {
-            // ADD JAR
-            callAddJar((AddJarOperation) operation);
-        } else if (operation instanceof RemoveJarOperation) {
-            // REMOVE JAR
-            callRemoveJar((RemoveJarOperation) operation);
-        } else if (operation instanceof ShowJarsOperation) {
-            // SHOW JARS
-            callShowJars();
-        } else if (operation instanceof ShowCreateTableOperation) {
-            // SHOW CREATE TABLE
-            callShowCreateTable((ShowCreateTableOperation) operation);
-        } else {
-            // fallback to default implementation
-            executeOperation(operation);
+        final Thread thread = Thread.currentThread();
```

Comment:
How about move the logic about register  handler to the `executeStatement` part. We may add more commands in the future and the block will be larger and lager.

```
           operation.ifPresent(
                    op -> {
                        final Thread thread = Thread.currentThread();
                        final Terminal.SignalHandler previousHandler =
                                terminal.handle(
                                        Terminal.Signal.INT, (signal) -> thread.interrupt());
                        try {
                            callOperation(op, executionMode);
                        } finally {
                            if (previousHandler != null) {
                                terminal.handle(Terminal.Signal.INT, previousHandler);
                            }
                        }
                    });
```

Reply:
Seems much more reasonable,  fix it later.